lib/html/pipeline/bbcode_filter.rb

```diff
@@ -5,14 +5,22 @@ class Pipeline
     class BbcodeFilter < TextFilter
       def initialize(text, context = {}, result = nil)
         super text, context, result
-        @context = context
-        @text = @text.gsub "\r", ''
       end
 
       def call
-        html = BBCoder.new(@text).to_html.gsub(/\n|\r\n/, '<br />')
+        html = BBCoder.new(@text).to_html
+        html = remove_url_link_contents(html)
+        html.gsub('<br>', '')
         html.rstrip!
-        "<p>#{html}</p>"
+        html
+      end
+
+      def remove_url_link_contents(html)
+        doc = Nokogiri::HTML::DocumentFragment.parse(html)
+        doc.css('a').each do |link|
+          link.content = link.content.gsub(%r{https?://}, '')
+        end
+        doc.to_html
```

Comment:
The reason this is here is because the combination of bbcode and markdown butchers the generated markup for something like:

```
[url]http://yahoo.com[/url]  =>  <a href="http://yahoo.com"><a href="http://yahoo.com">http://yahoo.com</a></a>
```

With this method it'll find the contents of `<a>` tags and remove the `https?://` before it hits the markdown filter.